lib/compiler.js

```diff
@@ -513,16 +514,8 @@ function buildComponentInstructions(rootDir, componentConfig) {
   return Object.keys(componentConfig)
     .filter(function(name) { return !!componentConfig[name]; })
     .map(function(name) {
-      var sourceFile;
-      if (name.indexOf('./') === 0 || name.indexOf('../') === 0) {
-        // Relative path
-        sourceFile = path.resolve(rootDir, name);
-      } else {
-        sourceFile = require.resolve(name);
-      }
-
       return {
-        sourceFile: sourceFile,
+        sourceFile: resolveAppScriptPath(rootDir, name),
```

Comment:
Before this change, a component path "mymod/file.js" was never resolved as relative, i.e. "{appRootDir}/mymod/file.js".

Is there a reasonably easy way how to preserve this behaviour? For example, can we add an `options` argument to `resolveAppScriptPath` that will be passed all the way down to `tryResolveAppPath` and which will control whether values that don't start with "./" or "../" are resolved as relative or not?

Example usage here:

```js
sourceFile: resolveAppScriptPath(rootDir, name, { strict: true })
```

Thoughts?